packages/components/src/drop-zone/style.scss

```diff
@@ -5,31 +5,69 @@
 	bottom: 0;
 	left: 0;
 	z-index: z-index(".components-drop-zone");
-	visibility: hidden;
-	opacity: 0;
 	border-radius: $radius-block-ui;
 
+	.components-drop-zone__content {
+		position: absolute;
+		top: 0;
+		bottom: 0;
+		left: 0;
+		right: 0;
+		height: 100%;
+		width: 100%;
+		display: flex;
+		background-color: $components-color-accent;
+		align-items: center;
+		justify-content: center;
+		z-index: z-index(".components-drop-zone__content");
+		text-align: center;
+		color: $white;
+		opacity: 0;
```

Comment:
If you look at the initial visual state here and for the `__content-inner` you might notice they might be redundant with the ones defined in the `.inactive` class below and you're right. However, it's needed to handle the drop scenario without making the expressions in the component more complicated:

1. In the `shouldActivate` expression, the `isDraggingOverDocument` seems to be needed, or the drag-in sometimes is not detected.
2. However, you can't just negate that expression, as it doesn't work well to detect the drag-out. For dragging-out, we should only check if `! isDraggingOverElement`.
3. This means that `is-active` stays applied even if you drag-out (because it includes the `isDraggingOverDocument`, which remains `true`). That's okay for dragging out as long as `inactive` is applied, which it is because when you drag out, `isDraggingOverElement` changes to `false`.
4. However, when you drop inside the zone, then `inactive` is not applied. We could change the expression there to also apply it, but it's simpler to just then let it fallback to the property here, which it does, because when you drop, then `is-active` is not applied (nor `inactive`).

Reply:
Ended up renaming the class to `has-dragged-out` as it felt like it better represented the meaning behind it. The way I understand this change, is that it allows removing the conditional rendering of the drop zone, and it allows us to achieve the same effect just using CSS classes.

I also managed to remove some CSS associated to the `inactive` classname by applying the active styles only when `&.is-active:not(.has-dragged-out)`.